### PR TITLE
DAO Facade

### PR DESCRIPTION
**Pull request by momoasalu**
Created DAO facade + delegated some tasks

### CODE REVIEW

File: src/data_access/DataAccessObjectFacade.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        userDataAccessObject.setResponse(response);
+    }
+
+    @Override
+    public String getActivePromptText() {
+        return null;
+    }
+
+    @Override
+    public UUID getActivePromptId() {
```
Comment (chowdberry) [inline src/data_access/DataAccessObjectFacade.java:37]: Likely do not need to implement this method within the UserDAO, can use the PromptDAO for this method.

**Review by chowdberry — APPROVED** [review]
- Good idea to implement a facade design pattern, we can probably get rid of some of the User DAO methods that overlap with the Prompt DAO methods.

File: src/data_access/DataAccessObjectFacade.java (hunk 1 above)
Comment (momoasalu) [reply src/data_access/DataAccessObjectFacade.java:37]: That specific method is delegating to just the prompt DAO, not the User DAO! I just didn’t have the code for the Prompt DAO at the time so it returns null.

**Comment by NimratGrewal** [comment]
just want to make sure, this facade pattern currently only calls the methods from the UserDataAccessObject and not the PromptDAO? are we also creating one for PromptDAO?

**Comment by momoasalu** [comment]
> just want to make sure, this facade pattern currently only calls the methods from the UserDataAccessObject and not the PromptDAO? are we also creating one for PromptDAO?

It will call for both of them but I only had the user dao when I created the branch. I will work on delegating to the prompt DAO after.

**Review by NimratGrewal — APPROVED** [review]

**Pull request merged**